packages/salesforcedx-utils-vscode/src/cli/orgCreateResultParser.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+export interface OrgCreateSuccessResult {
+  status: number;
+  result: {
+    orgId: string;
+    username: string;
+  };
+}
+
+export interface OrgCreateErrorResult {
+  status: number;
+  name: string;
+  message: string;
+  exitCode: number;
+  commandName: string;
+  stack: string;
+  warnings: any[];
+}
+
+export class OrgCreateResultParser {
+  private response: any;
+
+  constructor(stdout: string) {
+    try {
+      const sanitized = stdout.substring(
+        stdout.indexOf('{'),
+        stdout.lastIndexOf('}') + 1
+      );
+      this.response = JSON.parse(sanitized);
+    } catch (e) {
+      const err = new Error('Error parsing org create result');
+      err.name = 'OrgCreateParserFail';
+      throw err;
+    }
+  }
+
+  public createIsSuccessful(): boolean {
+    return this.response && this.response.status === 0 ? true : false;
+  }
+
+  public getResult(): OrgCreateSuccessResult | OrgCreateErrorResult {
```

Comment:
Originally I wanted to use the success result data to set the org type in the execute method of the command. I didn't end up doing that because we currently only run the command for scratch orgs but we'll soon have to since it's also going to support creating sandboxes. 
Do we want to keep this around or should I change it to exclusively returning `OrgCreateErrorResult` ? I'm ok with both options.

Reply:
I think it's okay the way you have it now. You left a note to address this when we support creating sandboxes.